### ISSUE TITLE
v0.2.0

### ISSUE THREAD

**Issue by esbullington** [esbullington/react-d3#28]
To do for v0.2.0:

1) Add maybe two more chart types: a simple gauge chart (esbullington) and  scatter (yang-wei) 
2) Improve documentation (esbullington working on it)
3) Add legends (esbullington working on it)
4) Get the code base cleaned up, and start using jshint (esbullington working on it)
5) Factor out the axes (almost done by esbullington in #27 )
6) Add some simple mouseover animations for the line, bar, and scatter charts  (yang-wei)

**Comment by berrytj** [comment]
Hi @esbullington, I'm thinking of adding labels to the linechart – is this something you've already worked on?

**Comment by berrytj** [comment]
Not sure if that's what you mean by "legend", above.

**Comment by esbullington** [comment]
Hi @berrytj thanks for your input and contribution (I'll check your PR right now).  Yes, I'm working on both a [legend](http://bl.ocks.org/ZJONSSON/3918369) and [label](http://www.d3noob.org/2012/12/adding-axis-labels-to-d3js-graph.html) components right now.  Should have it up today or maybe tomorrow morning.

**Comment by yang-wei** [comment]
hey @esbullington regarding to your todo 6) do you have any idea on mind? Refer to [nvd3](http://nvd3.org/examples/scatter.html) examples, "mouse over" will
 * change the color of dots to darker color
 * increase the size of dots
 * show extra informations on tooltips
Since we're only on early stage so do you think that we can implement the first 2 animation(color and size)

**Comment by esbullington** [comment]
hey @yang-wei that's exactly what I have in mind: the first two things mouse over animations would be great for this release.

Tooltips are more complicated so we can save that for a later release.

**Comment by yang-wei** [comment]
cool ! I'm willing to take over that =) @esbullington

**Comment by esbullington** [comment]
That'd be great, @yang-wei  I'm finishing up the legends and also working on automating the build, release and npm publish process a bit more, as well as documentation.

I've restarted the legend several times because I've not been happy with the appearance.  But I think I found a look I like.

I also plan on finishing up the gauge chart before 0.2.0.

**Comment by yang-wei** [comment]
@esbullington Hey regarding to the animation, I have question to ask. Because most of our `fill`(color) and `r`(radius) are properties, so one way I can think of is to set up `onMouseOver` event on them, when the event is trigger we'll pass it to the top parent (which i think is `<Chart />`) and re-render them using `setProps()`. 

This will be a ugly hack =(　Becasue I have to pass a handler for each parent upper than the child. If you need more concrete example, I will add a fiddle or branch. Do you have any idea?

**Comment by esbullington** [comment]
@yang-wei You're right.  I think in this case, where we're dealing with strictly local, ephemeral state, using React `state` is OK.  This is the one exception I can think of to my "no React state" rule-of-thumb, as long as we keep the state confined to a single component (no callbacks passed up to parents, for example*).  For the state we'll use to control data frame animations, I want to use an immutable cursor passed down the component hierarchy.

Ideally, it'd be nice to just pass down state via props all the time, but you're correct that trying to do this in the way you describe is far too complex when we could just do:

```
var Chart = React.createClass({

  getDefaultState: function() {
    // setting state with props is not an antipattern here
    // since we're just setting initial width for our mouseover effect
    // see bottom of the page: 
    // http://facebook.github.io/react/tips/props-in-getInitialState-as-anti-pattern.html
    return {
      circleWidth: this.props.width
    }
  },

  animateCircle: function(
    this.setState({circleWidth: this.state.circleWidth+5}); // or whatever
  },

  render: function() {
    return <ChartCircle onMouseOver={this.animateCircle} onMouseOut={this.restoreCircle} width={this.state.circleWidth} />;
  }

});
```

\*  The reason I want to keep even ephemeral state completely confined to a single component is that I've been thinking a lot more about where I want to go with this project.  And my conclusion is that I want to move in the direction of a) continuing to offer an easy, single component API for different chart types, but at the same time b) also take more care to make our internal components easily usable and reusable, and export all of them for the end user.  I'll be writing more about this in a blog post soon.

**Comment by yang-wei** [comment]
Yeah this is perfect for me =) I will give it a try when I wake up next morning. Thanks for everything, learned so much today ! @esbullington 

Can't wait for your blog post !

**Comment by esbullington** [comment]
Sounds good @yang-wei  Thanks again!

Also, I may be committing a lot of changes to different components today, so I'd recommend pulling down the changes in the morning before you commit any code.  Or else check and if needed rebase off of master if you've already started work on a feature branch.

**Comment by yang-wei** [comment]
cool, will check them out @esbullington !

**Comment by esbullington** [comment]
Merged in #88.
